src/yunohost/certificate.py

```diff
@@ -858,12 +858,40 @@ def _domain_is_accessible_through_HTTP(ip, domain):
         requests.head("http://" + ip, headers={"Host": domain}, timeout=10)
     except requests.exceptions.Timeout as e:
         logger.warning(m18n.n('certmanager_http_check_timeout', domain=domain, ip=ip))
-        return False
     except Exception as e:
         logger.debug("Couldn't reach domain '%s' by requesting this ip '%s' because: %s" % (domain, ip, e))
+    else:
+        return True
+
+    # fallback on external YunoHost service because http loopback might be
+    # broken (this is actually way more common than excepted and LE doesn't
+    # care about local loopback, it wants outside reachability)
+
+    try:
+        response = requests.post("https://http-check.yunohost.org/check/", data={"domain": domain})
+    except Exception as e:
+        logger.warning("Error while trying to reach https://http-check.yunohost.org/check/ because: %s" % (e))
+        return False
+
+    if response.status_code == 200:
+        return True
+
+    # this mean that the domain couldn't be reached with a known exception
+    # (timeout for now)
+    # also talks about teapot
+    if response.status_code == 418:
```

Comment:
Teapots are awesome :+1: but I can't see any code in the server.py returning 418 ? Or am I missing something :thinking:

Reply:
Oupsi I forgot to push >_<

Here it is https://github.com/YunoHost/check-http/commit/7d868ec5ddcbf37018548464ec28163d6885edcb

Reply:
Also I'm using 418 because I couldn't find any other code that would be "200 but actually it's False" :/